### DISCUSSION TITLE
facing issue with deadpool redis

### DISCUSSION THREAD

**Discussion by rapidshaurya** [bikeshedder/deadpool#314] [Q&A] [answered]
When using futures::executor::block_on() function with pool.get(), the code is still waiting for a long time, and nothing is happening.

**Comment by bikeshedder** [comment, marked as answer]
Deadpool itself is executor agnostic (except for timeouts). You're using `deadpool-redis` which does call into the `redis` library when creating new connections. The `redis` crate only supports `tokio` or `async-std`.

I don't think it is possilble to use `futures::executor` with the `redis` crate. And therefore you can't use `futures::executor` with `deadpool-redis` either.

This should work (using tokio):

```rust
use tokio::runtime::Handle;

let handle = Handle::current();
let conn = handle.block_on(pool.get());
```

Though I really wonder why you want to use deadpool in sync code for fetching an async redis connection object. The only way to use the obtained connection object is via another `block_on` call. You'd be better of putting all the code inside an `async` function.
